freva-rest/src/freva_rest/auth.py

```diff
@@ -178,35 +96,35 @@ async def refresh_access_token(
             help="The client id that is used for the refresh token",
         ),
     ] = None,
+    client_secret: Annotated[
+        Optional[str],
+        Form(
+            title="Client secret",
+            alias="client_secret",
+            help="The client secret that is used for the refresh token",
+        ),
+    ] = None,
 ) -> Token:
-    """Renew the login token with help of a refresh token."""
-    return await get_token(
-        client_id=client_id or "", refresh_token=refresh_token
-    )
-
-
-@app.get("/api/auth/v2/status", tags=["Authentication"])
-async def get_token_status(
-    current_user: TokenPayload = Depends(get_user_status),
-) -> TokenPayload:
-    """Check the status of an access token."""
-    return current_user
-
-
-@app.post("/api/auth/v2/token/", response_model=Token, tags=["Authentication"])
-async def login_for_access_token(
-    form_data: KeycloakOAuth2RequestForm = Depends(),
-) -> Token:
-    """Create an new login token from a username and password.
-
-    You should either set a username and password or an existing refresh token.
-    You can also set the client_id. Client id's are configured to gain access,
-    specific access for certain users. If you don't set the client_id, the
-    default id will be chosen.
-
-    """
-    return await get_token(
-        client_id=form_data.client_id or "",
-        username=form_data.username,
-        password=form_data.password,
-    )
+    """Interact with the openID connect endpoint for client authentication."""
+    data: Dict[str, Optional[str]] = {
+        "client_id": (client_id or "").replace("None", "") or server_config.oidc_client,
+        "client_secret": client_secret or os.getenv("OIDC_CLIENT_SECRET", ""),
+        "grant_type": grant_type,
+    }
+    if grant_type == "password":
+        data["password"] = password
+        data["username"] = username
+    else:
+        data["refresh_token"] = refresh_token
+    async with aiohttp.ClientSession(timeout=TIMEOUT) as client:
```

Comment:
another point, in data-loader we perfectly consider ssl certs to secure the connection but here we don't use it here. But the problem which is gonna happen, i think it would be in `dev`, which we need to consider an env for this:


```python
ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
ssl_context.load_cert_chain('/path/to/cert.pem', keyfile='/path/to/key.pem')
async with aiohttp.ClientSession(timeout=TIMEOUT, connector=aiohttp.TCPConnector(ssl=ssl_context)) as client:
```
which `/path/to/cert.pem` and `/path/to/key.pem` can comes from env. because we use them in main function of cli of dataportal as well ...

https://medium.com/@mariovanrooij/adding-https-to-fastapi-ad5e0f9e084e

Reply:
I am not sure I understand this. The data-loader isn't involved here.

The ssl connection to the keycloak shouldn't be based on a self-signed cert. 

On the other hand for the redis connection in the data-loader, we want to use a self-signed cert to make sure that only hosts that hold the private key can make connections to the redis host.